exist-core/src/main/java/org/exist/xquery/XQueryContext.java

```diff
@@ -1569,8 +1569,8 @@ Module initBuiltInModule(final String namespaceURI, final String moduleClass) {
             if (LOG.isDebugEnabled()) {
                 LOG.debug("module " + module.getNamespaceURI() + " loaded successfully.");
             }
-        } catch (final ClassNotFoundException e) {
-            LOG.warn("module class " + moduleClass + " not found. Skipping...");
+        } catch (final ClassNotFoundException cnfe) {
+            LOG.warn("module class " + moduleClass + " not found. Skipping...", cnfe);
```

Comment:
@gmella I would like to avoid putting the full stack-trace in the log, as this is at `WARN` level.
Rather, I think that `cnfe.getMessage()` might contain enough detail of the missing class name?

Reply:
No experience to say if such case may be considered as an error and add such stack trace.
`grep -re "catch*ClassNotFoundException" -A3 src` shows us some other cases that do not throw any exception with non consistent rules. 

In any case showing the root Cause would be the best compromise to help understanding. This may be done using a helper function for summarizing exception causes ?
We use such [kind of code](https://github.com/JMMC-OpenDev/jMCS/blob/764395f1c0d67ea36b0b70ba68889e29456d4ec7/src/main/java/fr/jmmc/jmcs/gui/component/MessagePane.java#L153) to report feedback from our applications.

Reply:
Any idea if we need that complexity, or if `getMessage()` would suffice here?

Reply:
yes adding `getMessage()` probably is a small enhancement here. But in any case such report would appear after previous warning log that could explain this classnotfoundexception. We can ignore this PR. 

Now I understand that my analysis was not clear because I focused more on the error than the warning as shown in the next log extract:

> 
> 2019-09-11 11:29:46,407 [qtp1366811528-40] WARN  (XQueryContext.java [initBuiltInModule]:1573) - module class fr.jmmc.exist.OIExplorerModule not found. Skipping... 
> 2019-09-11 11:29:46,407 [qtp1366811528-40] ERROR (XQueryServlet.java [process]:559) - Cannot compile xquery: err:XPST0081 error found while loading module jmmc-oiexplorer: Error while loading module /db/apps/jmmc-resources/content/jmmc-oiexplorer.xql: Invalid qname oi:to-xml. No namespace defined for prefix oi. QName is invalid: INVALID_PREFIX 
> ...